test_runner/regress/test_remote_storage.py

```diff
@@ -436,8 +436,8 @@ def assert_compacted_and_uploads_queued():
 
     assert not timeline_path.exists()
 
-    # timeline deletion should kill ongoing uploads
-    assert get_queued_count(file_kind="index", op_kind="upload") == 0
+    # timeline deletion should kill ongoing uploads, so, the metric will be gone
+    assert get_queued_count(file_kind="index", op_kind="upload") is None
```

Comment:
`get_queued_count` returns int()
so this needs some fix

I'd expect that it will fail on `assert matches` (line 256), but it isn't. 
Looks like we don't drop the metric properly

Reply:
> get_queued_count returns int()
> so this needs some fix

Right.

> I'd expect that it will fail on assert matches (line 256), but it isn't.
> Looks like we don't drop the metric properly

It's a different nested function definition.

One is in line 249.
Another is in line 380.

Different test cases as well.

For the one in 249, we don't detach the tenant delete the timeline, so, `!= None` is expected.

For the one in 380, we call it once while the timeline is there and once after we've deleted the timeline.
So, it needs to be able to handle both.

Will fix it.